src/dfcx_scrapi/tools/search_util.py

```diff
@@ -72,11 +81,37 @@ def __init__(
         self.intents_map = None
         if agent_id:
             self.agent_id = agent_id
-            self.flow_map = self.flows.get_flows_map(
+            self.flows_map = self.flows.get_flows_map(
                 agent_id=agent_id, reverse=True
             )
             self.intents_map = self.intents.get_intents_map(agent_id)
             self.client_options = self._set_region(agent_id)
+            self.flow_data = self.get_all_flow_data()
+            self.page_data = self.get_all_page_data()
+            self.route_group_data = self.get_all_route_group_data()
+
+    def get_all_flow_data(self):
+        flow_data = {}
+        flow_list = self.flows.list_flows(self.agent_id)
+        for flow in flow_list:
+            flow_data[flow.name] = flow
+        return flow_data
+
+    def get_all_page_data(self):
+        page_data = {}
+        for flow_id in self.flows_map.values():
```

Comment:
@SeanScripts @skyyeat 

I think you're unnecessarily looping through the `self.flows_map` twice across these two methods:
- `get_all_page_data`
- `get_all_route_group_data`

Since you have to loop through the flow to get to _both_ the Page and Route Group data, you should build _both_ sets of data simultaneously as you go through `self.flows_map`. 

Since you are already storing the data in `self.page_data` and `self.route_group_data` you can easily update these from within a single method.